changelog/4058.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixes a bug in fido_factory to allow  path="./" in fido.fetch().
```

Comment:
The codestyle check thinks there is a stray end of line here: https://dev.azure.com/sunpy/sunpy/_build/results?buildId=7271&view=logs&j=4cdc857c-0d48-50f8-2772-3b048c8bb2f3&t=fe62049c-f9e9-5b31-8bb6-a78aae0c6ebe&l=40